src/main/java/com/google/sps/servlets/utils/Statistics.java

```diff
@@ -86,22 +126,31 @@ private void setAggregateScores(List<Double> sentimentScores) {
       // update the score pointer
       updatingScoreIdx = scoreIdx;
     }
+
+    // If the categorization allows overflow, add the value larger than upper end to the last
+    // interval
+    if (overFlow && updatingScoreIdx < sentimentValues.size()) {
+      aggregateValues.put(
+          new Range(tempPoint.subtract(interval), tempPoint),
+          aggregateValues.get(new Range(tempPoint.subtract(interval), tempPoint))
+              + (sentimentValues.size() - updatingScoreIdx));
+    }
+    return aggregateValues;
   }
 
   /**
-   * Set the average score of given sentiment scores. Throws an runtime exception if the average
-   * score is not valid or none of the sentiment scores is valid.
+   * Set the average value of given sentiment values.
    *
-   * @param sentimentScores a list of score values from -1.0 to 1.0
+   * @param sentimentValues a list of values to calculate the average for
+   * @return the average value of sentimentValues
    */
-  private void setAverageScore(List<Double> sentimentScores) throws RuntimeException {
-    averageScore =
-        sentimentScores.stream()
-            .mapToDouble(i -> i)
-            .average()
-            .orElseThrow(
-                () ->
-                    new RuntimeException(
-                        "Unable to calculate sentiment average due to empty input list."));
+  private double getAggregateAvg(List<Double> sentimentValues) {
```

Comment:
NIT:

Change to `getAggregateAverage`. We already use the term `average` in a lot of places, and it doesn't make sense to switch to using `avg` in some places (even if it saves us a few precious characters).